src/picongpu/include/plugins/hdf5/NDScalars.hpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2016 Alexander Grund
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "plugins/hdf5/HDF5Writer.def"
+#include "traits/PICToSplash.hpp"
+#include "Environment.hpp"
+
+namespace picongpu {
+namespace hdf5 {
+
+/* Functors for reading and writing ND scalar fields with N=simDim
```

Comment:
just a side note: even if `simDim` is 3 it does not mean one needs to do the domain decomposition in 3 dimensions. In the context of ID providers this is a bit irritating, but ... ook.